sdk/android/test/sdk.testapp/src/main/java/com/microsoft/windowsazure/mobileservices/sdk/testapp/test/MobileServiceSyncTableTests.java

```diff
@@ -538,6 +537,58 @@ public void testIncrementalPullSaveLastUpdatedAtDate() throws MalformedURLExcept
         assertEquals(updatedAt2, stringMaxUpdatedDate);
     }
 
+    public void testincrementalSyncUsesSkipWhenMaxUpdateAtIsSame() throws MalformedURLException, InterruptedException, ExecutionException, MobileServiceException {
+
+        MobileServiceLocalStoreMock store = new MobileServiceLocalStoreMock();
+        ServiceFilterContainer serviceFilterContainer = new ServiceFilterContainer();
+        String queryKey = "QueryKey";
+        String incrementalPullStrategyTable = "__incrementalPullData";
+
+        MobileServiceClient client = new MobileServiceClient(appUrl, appKey, getInstrumentation().getTargetContext());
+
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        Date d = new Date();
+        String updatedAt1 = sdf.format(d);
+        d.setTime(d.getTime() + 1000L);
+        String updatedAt2 = sdf.format(d);
+
+        client = client.withFilter(getTestFilter(serviceFilterContainer, false,
+                "{\"count\":\"6\",\"results\":" +
```

Comment:
why are you mocking a count?  A pull shouldn't be asking for inlinecount on the query.  Also, whys it returning 6?  Does the Android filter logic work different than I am used to with managed & iOS?